packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -714,29 +999,50 @@ class CupertinoActionSheetAction extends StatelessWidget {
   /// Typically a [Text] widget.
   final Widget child;
 
+  @override
+  State<CupertinoActionSheetAction> createState() => _CupertinoActionSheetActionState();
+}
+
+class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
+    implements _ActionSheetDragAvatar {
+  // |_ActionSheetDragAvatar|
+  @override
```

Comment:
It is unfortunate we need to split the background from this widget...

Reply:
Yeah, I wish we didn't have to. 

The problem with this separation actually comes from the cancel button. Our current sample code uses the same `CupertinoActionSheetAction` for the cancel button, but the cancel button uses a different background color. To  make `CupertinoActionSheetAction` paint its own background, it has to at least receive a `.isCancel = true` property, or somehow receive an inherited data from the parent to know that it's a cancel button (which is even uglier).